cf_xarray/accessor.py

```diff
@@ -168,6 +177,57 @@ def _get_groupby_time_accessor(var: Union[DataArray, Dataset], key: str) -> List
         return []
 
 
+def _get_custom_criteria(
+    obj: Union[DataArray, Dataset], key: str, criteria=None
+) -> List[str]:
+    """
+    Translate from axis, coord, or custom name to variable name optionally
+    using ``custom_criteria``
+
+    Parameters
+    ----------
+    obj : DataArray, Dataset
+        DataArray belonging to the coordinate to be checked
+    key : str, ["X", "Y", "Z", "T", "longitude", "latitude", "vertical", "time"], or a key from user-defined custom_criteria
+        key to check for.
+    criteria : dict, optional
+        Criteria to use to map from variable to attributes describing the
+        variable. An example is coordinate_criteria which maps coordinates to
+        their attributes and attribute values. If user has defined
+        custom_criteria, this will be used by default.
+
+    Returns
+    -------
+    List[str], Variable name(s) in parent xarray object that matches axis, coordinate, or custom `key`
+
+    """
+
+    if isinstance(obj, DataArray):
+        obj = obj._to_temp_dataset()
+
+    if criteria is None:
+        if not OPTIONS["custom_criteria"]:
+            return []
+        criteria = OPTIONS["custom_criteria"]
+
+    if criteria is not None:
+        criteria = always_iterable(criteria, allowed=(tuple, list, set))
+
+    criteria = ChainMap(*criteria)
+
+    results: Set = set()
+    if key in criteria:
+        for criterion, expected in criteria[key].items():
+            for var in obj.variables:
+                # Treat expected as regex
+                if re.match("|".join(expected), obj[var].attrs.get(criterion, "")):
+                    results.update((var,))
+                # also check name specifically since not in attributes
+                elif criterion == "name" and var in expected:
```

Comment:
I bet someone will want the regex version here too.

Reply:
Do you mean someone would want regex in the name check? I removed it because I thought maybe for "name" comparison, people would want it to be exact. The problem for me that prompted that was I had variables "wind_speed" and "wind_speed_of_gust", and the check for the first was picking up the second. Or is there a better way to separate?

Reply:
oh so that's a problem for the `standard_name` checks too then

Should we instead have
```suggestion
    my_custom_criteria = {
        "ssh": {
            "standard_name": {"regex": ("sea_surface_elev*",)},
            "name": {"exact": ("sea_surface_elevation",)},  # variable name
        },
    }
```

so there's direct control of regex vs exact matches

Reply:
Yes that is a nice way to make it explicit. Doing this here: https://github.com/xarray-contrib/cf-xarray/pull/234/commits/a6a527764e0fd0fcfd3811c810be42fd21482c7f